Infrastructure/ARM/Deploy.json

```diff
@@ -0,0 +1,259 @@
+                        "lenRoles": "[length(parameters('roles'))]",
+                        "lenTotal": "[mul(variables('lenPrincipalIds'), variables('lenRoles'))]"
+                    },
+                    "resources": [
+                        {
+                            "copy": {
+                                "count": "[variables('lenTotal')]",
+                                "name": "pairs"
+                            },
+                            "type": "Microsoft.Authorization/roleAssignments",
+                            "apiVersion": "2022-04-01",
+                            "name": "[guid(variables('_roles')[parameters('roles')[div(copyIndex(), variables('lenPrincipalIds'))]], parameters('principalId')[mod(copyIndex(), variables('lenPrincipalIds'))])]",
+                            "properties": {
+                                "principalId": "[parameters('principalId')[mod(copyIndex(), variables('lenPrincipalIds'))]]",
+                                "roleDefinitionId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions',variables('_roles')[parameters('roles')[div(copyIndex(), variables('lenPrincipalIds'))]])]"
+                            }
+                        }
+                    ]
+
+                },
+                "parameters": {
+                    "principalId": {
+                        "value": [
+                            "[reference(variables('identityId')).principalId]"
+                        ]
+                    },
+                    "roles": {
+                        "value": [
+                            "Key Vault Secrets User",
+                            "Storage Account Queue Contributor",
+                            "Storage Account Blob Data Owner",
+                            "Storage Account Contributor",
+                            "Storage Account Table Contributor"
+                        ]
+                    }
+                }
+            }
+        },
+        {
+            "name": "[variables('storageAccountName')]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2022-05-01",
+            "location": "[variables('location')]",
+            "sku": { "name": "Standard_LRS" },
+            "kind": "StorageV2",
+            "properties": {
+                "isHnsEnabled": true,
+                "isLocalUserEnabled": true,
+                "isSftpEnabled": true
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts/fileServices/shares",
+            "apiVersion": "2022-09-01",
+            "name": "[concat(variables('storageAccountName'), '/default/', variables('shortName'), 'sha')]",
+             "dependsOn": [
+                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+            ]
+            },
+        {
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2020-02-02",
```

Comment:
this version is almost deprecated and recently AI requires associations to LAW (log analytics workspace).

please use LAW-linked AI, and assume LAW.Id to be provided externally as parameter.